Makefile

```diff
@@ -23,6 +23,12 @@ LDFLAGS_COMMON := -X main.gitCommit=$(COMMIT) -X main.version=$(VERSION)
 
 GOARCH := $(shell $(GO) env GOARCH)
 
+# Allow Go 1.22.4+ to compile nsenter. This is ugly.
+GO122PATCH := $(shell $(GO) version | sed -n 's/.* go1\.22.\([0-9]*\) .*/\1/p')
+ifeq ($(shell test $(GO122PATCH) -ge 4 && echo yes), yes)
+	export CGO_CPPFLAGS+="-D GO1_22_WORKS=1"
+endif
```

Comment:
I guess we can now just block Go 1.22.0-1.22.3 ?

Reply:
You mean, block it from the top-level Makefile, and remove nsenter_go122 entirely? Makes sense

Reply:
Yes

Reply:
But for some libct users may need this file?

Reply:
How about write some code to support all go 1.22 versions?

Reply:
Please see #4310, feel free to close it if we don't want to support go 1.22.3, I think if we can support all go versions, it will be better.

Reply:
Let's just skip 1.22.3, for simplicity

Reply:
But for libct users, such as hashicorp/nomad, if they don’t write such method in Makefile, they will get an error even in go 1.22.4.

Reply:
OTOH we are checking glibc version in nsenter_go122 and it's somewhat hard to do from the Makefile.

Reply:
Leaving the current hack in place, added some extra comments.

Reply:
> if we don't want to support go 1.22.3, I think if we can support all go versions, it will be better.

To me it doesn't make much sense to support those older versions. We can just require Go 1.22 >= 1.22.4, it's a decent requirement.

> But for libct users, such as hashicorp/nomad, if they don’t write such method in Makefile, they will get an error even in go 1.22.4.

That's right. 

Ideally, I'd like to do it all in one place -- that is, `nsenter_go122.go`. Unfortunately, I can't find a way to determine go's patch version from there.

Reply:
> I can't find a way to determine go's patch version from there.

... during compile time.

Reply:
Is it so bad to not handle the case at all, then? I mean, not do anything for go 1.22, just document in the function/package that if using go 1.22, you need at least go 1.22.4 (or glibc and explain it all) and be done with it.

Go 1.22.4 also includes security fixes, so hopefully people upgrade faster.

A runc release will at least take 1-2 more weeks I think, so go 1.22.4 should be out for a few weeks already by then.

Reply:
IMHO, I'd leave this PR without handling the 1.22 case, we can discuss what to do for 1.22 #4310

Reply:
I agree, because go 1.22.4 has released, we think that people should swith to go 1.22.4, they will not use go 1.22.3, especially when running CI.

Reply:
Yeah, maybe we should just stop blacklisting go and instead add a note that for Go 1.22 one should use Go 1.22.4+.